src/AppBundle/Model/SlotCollectionDecorator.php

```diff
@@ -98,6 +98,51 @@ public function getSlotsByType()
         return $slotsByType;
     }
 
+    public static function sortByCycleOrder($s1, $s2)
+    {
+        return intval($s1->getCard()->getCode()) - intval($s2->getCard()->getCode());
+    }
+
+    public static function sortByCycleOrder_deluxeAfter($s1, $s2)
+    {
+        // Compare cycle first. Cycle size should be enough for the moment. Core set will be first anyway.
+
+        // If at least one is a core set slot, then ordering by code will be fine
+        $comparingCore = $s1->getCard()->getPack()->getCycle()->getPosition() == 1 || $s2->getCard()->getPack()->getCycle()->getPosition() == 1;
+
+        // If the packs have same size (both non-deluxe, or both deluxe), then ordering by code will be fine
+        $sameSize = $s1->getCard()->getPack()->getCycle()->getSize() == $s2->getCard()->getPack()->getCycle()->getSize();
+
+
+        if ($comparingCore || $sameSize)
+            // Normal ordering
+            return intval($s1->getCard()->getCode()) - intval($s2->getCard()->getCode());
+        else
+            // Cycle with size 6 (non-deluxe) should be first
+            return intval($s2->getCard()->getPack()->getCycle()->getSize()) - intval($s1->getCard()->getPack()->getCycle()->getSize());
+    }
+
+
+    public function getSlotsByCycleOrder($deluxeAfter)
+    {
+        $slots_array = [];
+        foreach ($this->slots as $slot){
+            $slots_array[] = $slot;
+        }
+
+        if (!$deluxeAfter)
+            usort($slots_array, array("AppBundle\Model\SlotCollectionDecorator", "sortByCycleOrder"));
+        else
+            usort($slots_array, array("AppBundle\Model\SlotCollectionDecorator", "sortByCycleOrder_deluxeAfter"));
```

Comment:
as stated above, the callback function's name should be `sortByCycleOrderDeluxeAfter`

Reply:
Solved in 5a40eda332fa1946c3fd2bbf2db53f8c111ae898